src/Workspaces/CSharp/Portable/CodeGeneration/ParameterGenerator.cs

```diff
@@ -140,14 +141,34 @@ internal static ParameterSyntax GetParameter(IParameterSymbol p, CodeGenerationO
         }
 
         private static SyntaxList<AttributeListSyntax> GenerateAttributes(
-            IParameterSymbol parameter, bool isExplicit, CodeGenerationOptions options)
+            IParameterSymbol parameter, bool isExplicit, CodeGenerationOptions options, ref NullableAnnotation nullableAnnotation)
         {
             if (isExplicit)
             {
                 return default;
             }
 
             var attributes = parameter.GetAttributes();
+
+            if (parameter.Type.IsReferenceType)
+            {
+                var newAttributes = attributes.RemoveAll(IsAllowNullAttribute);
+                if (newAttributes.Length != attributes.Length)
+                {
+                    attributes = newAttributes;
+                    if (nullableAnnotation == NullableAnnotation.NotAnnotated)
+                    {
+                        nullableAnnotation = NullableAnnotation.Annotated;
+                    }
+                }
+            }
+            else if (parameter.Type.IsValueType)
+            {
+                attributes = parameter.NullableAnnotation == NullableAnnotation.Annotated
```

Comment:
I didn't think parameter.NullableAnnotation would be set in the struct case, because we reserved these only for reference types? @333fred?

Reply:
This was discussed in https://github.com/dotnet/roslyn/pull/38203#discussion_r336107314.

Reply:
Yep, flow analysis will tell you about NVTs as well.